_studio/shared/umc/codec/h264_dec/src/umc_h264_task_supplier.cpp

```diff
@@ -1212,11 +1212,13 @@ void MVC_Extension::ChooseLevelIdc(const H264SeqParamSetMVCExtension * extension
 
     m_level_idc = 0;
 
-    for (uint32_t i = 0; i <= extension->num_level_values_signalled_minus1; i++)
+    assert(extension->num_level_values_signalled_minus1 + 1 == extension->levelInfo.size());
+    for (uint32_t i = 0; i < extension->levelInfo.size(); i++)
```

Comment:
ok, I think all this scary construction is a good example of how stl algorithms can simplify things:
```
    for (const auto& levelInfo : extension->levelInfo)
    {
        assert((levelInfo.num_applicable_ops_minus1 + 1) == levelInfo.opsInfo.size());
        
        for (const auto& operationPoint : levelInfo.opsInfo)
        {            
            if (std::find_first_of(m_viewIDsList.begin(), m_viewIDsList.end(),
                                   operationPoint.applicable_op_target_view_id.begin(), operationPoint.applicable_op_target_view_id.end())
                        == m_viewIDsList.end())
                break;

            if (!m_level_idc || m_level_idc > levelInfo.level_idc)
            {
                m_level_idc = levelInfo.level_idc; // <- looks weird, we operate with levelInfo on upper level, we have assignment of loop invariant here?
            }    
        }
    }
```

Could you also find an answer on my question?

Reply:
find_first_of returns !end() one  element from m_viewIDsList contains in applicable_op_target_view_id,
but old code check that **all** elements of m_viewIDsList  contain in applicable_op_target_view_id

Reply:
yep, you are right... does the order of elements matters? can we sort them and just use ```std::includes``` ? The other variant is to construct two ```std::set``` and check that one is a subset of another (but I don't like this approach, because of overhead on sets construction)